RPS_GUI

Calerid opened the pull request:
I believe that this is a good candidate for merging with the base.

-Fixed issues with each function passing parameters. 
-Added on screen buttons versus the player using the console.
-Fixed issues with on screen information not updating prior to prompt interrupt of DOM repaint.
-Added round win/loss messages.

Please provide any feedback for things I may have missed or still need correcting. 

Thank you.

Calerid committed:
Fix paper beat scissors

Fixed a game breaking bug were scissors would be overcome by a large amount of paper. This was fatal,

Calerid committed:
Merge branch 'main' of github.com:calerid/My_Odin_Projects

Calerid committed:
Merge branch 'main' of github.com:calerid/My_Odin_Projects

Calerid committed:
Change index.html structure

Made several changes to the HTML structure so that the grid/sizing will be easier to read and work with. Script is now at the bottom of the page. Borrowed div structure from other projects on TOP

Calerid committed:
Add document node/elements

Added variables for the grid container div and a button which allows the user to set the grid size. Made several inline style changes to the h1 element.

Calerid committed:
Add new variables

Started work on adding a GUI for the app. Added new container in index.html, a coresponding style rule was created to set display grid. Finally created 3 new const variables within the js file. This is the foundation for adding a GUI way to play the game instead of using a terminal to play. I will leave the terminal method as well; however, add the GUI for most players

Calerid committed:
Add event listeners

Wrapped the three divs in buttons, to which I added event listeners. When finished these will trigger the game 'plays' instead of typing them into the promp

Calerid committed:
Add basic GUI

Implemented a basic GUI consisting of three buttons. These now trigger new setter functions which change the playerSelection variable, versus the player inputting into a prompt box. Attempts to add event listeners did not function as desired, the events are now processed on the HTML page via onclick. Removed startNewRound() calling on the player to set new inputs as well as several functions which called startNewRound(), now the page click initiates the next round by default

Calerid committed:
Fix result of game

Made several fixes to the results which were broken from the 'complete' version. In several events rocks were so fast they smashed through paper. Now paper safely wraps around rocks holding them in place

Calerid committed:
Merge branch 'RPS_GUI' of github.com:calerid/My_Odin_Projects into RPS_GUI

Calerid committed:
Add roundCount to console.log string literal

Added the output of each round into the string literal printed in the console.log(). This makes it easier to keep track of if the game goes over the limited round count in roundCounter().

Calerid committed:
Add html score elements

Added several objects to the HTML section. These will be used to show the score on the web page, preventing the user from having the use the console any longer. Console functions should remain in finished application for more than one way to use the game

Calerid committed:
Add textContent attribute to updateElements() variables on DOM elements.

Added the textContent to variables within the updateElements() function. This now updates the counts of scores, draws (needs work), and round count. Still needs work as unedfined shows in Draws. Round count also seems to go outside set limits now, unsure why

Calerid committed:
Move variable init

moved the variable initializing to the start of the program rather than within updateElements. Changes the variables to capitalized.

Calerid committed:
Fix issue with draws not displaying on page

Resolved an issue in which draws were not being passed on to the function which displayed the draw count on page. Added nobodyWins count being passed into the updateElements() function.
Round count of draws no longer displays as 'undefined'

Calerid committed:
Change css.style for button and #score

Changed css to make the buttons rounded, changed the style of the score box so that the font was larger and is now the overall color is red.

Calerid committed:
Move return roundCounter() to updateElements()

Moved around the function calls / returns so that each function/play now calls updateElements() to attempt updating the HTML before the round ends.
This was done because the 5th round never prints on the page prior to the prompt asking the player to start a new match.

Calerid committed:
Redo HTML & CSS structure

Reworked the HTML, adding the roundMessage id which will be used to display the round outcome messages. Reworked the CSS to make it more simplified, for the time being all text is in white, no outline.

Calerid committed:
Add counter variables for updateElements() function

Added local variables within the updateElements() function to then change the textContent / innerText of the scores.

Calerid committed:
Change button color

Changed the button color to blue.

Calerid committed:
Update const variable name

Updated the constant variable names for the round stats. This was done to improve readability and eliminate confusion.

Calerid committed:
mend

Calerid committed:
mend

Calerid committed:
Remove uneeded call and return statement

Removed the parameter passthrough of global variables which were not necessary. Also removed return statements for global variables. Now each function will work with the global variable instead
of passing through edited local variables.

Calerid committed:
mend

Calerid committed:
Fix bug with DOM repaint executing after prompt()

An error would occur in which the round count would be interrupted from the prompt() in newGamePrompt(). The solution was call setTimeout() on roundCounter(). This pause allows for the next repaint cycle to occur before the prompt() is called, interrupting the thread.

Calerid committed:
Add on screen win/loss message

Added messages to an HTML element which displays the round result message. These textContent updates are located in the three add functions addPlayerWins, addComputerWins(), and addNobodyWins().

PejmanPoh commented on rock-paper-scissors/index.html:
```diff
@@ -3,26 +3,26 @@
+        <button id="scissors" onclick="setScissors()">Scissors</button>
+      </div>
+      <div id="score">
+      <p>Player Score: <p id="playerScore">0</p></p>
```
Not sure if I've spoken to you about HTML semantics but if you've ever wondered why HTML has so many different tags that appear to do the same thing (e.g. <section> vs <div>), it's for screen readers and general parsing altogether. When the only thing that a person can read is basically the HTML code, semantically correct HTML is imperative to helping them understand the context.

This is a lot of text to tell you that a `<p>` tag inside a `<p>` tag is semantically incorrect. ([source](https://stackoverflow.com/questions/4291467/nesting-block-level-elements-inside-the-p-tag-right-or-wrong)).

Calerid replied:
Fixed.

PejmanPoh commented on rock-paper-scissors/scripts/rock-paper-scissors.js:
```diff
@@ -1,5 +1,14 @@
+const ROUND_MESSAGE = document.getElementById("roundMessage");
+
 let playerSelection = "";
 let computerSelection = "";
```
I want you to try removing these two lines and getting the program to still work. I won't say anything more than that but I believe this will help you better grasp parameter passing and variable scopes.

PejmanPoh reviewed (changes requested):
I think I've got a good challenge for you to take on, so I will leave my comments as just 1 random HTML comment and a 1 JS challenge. Let's see how you do!

Calerid committed:
Fix semantically incorrect HTML tags

Fixed a <p> in <p> tag that was poorly thought out. This is now dealt with entirely in the string literal and DOM assignment in updateElements().

Calerid committed:
Dummy commit / testing variables without init

let playerSelection and let computerSelection have been removed. The program still functions

Calerid committed:
Declared playerSelection and computerSelection as undefined

Declared both playerSelection and computerSelection on lines 17 and 18 as undefined. They will no longer be defined as empty strings and instead be assigned values at their respective functions.

Calerid committed:
Move computerSelection variable declaration and assignment to local scope'

computerSelection is now declared in startNewRound(), it calls on computerPlay() to return a random value as the assignment.

Calerid committed:
Declare computerSelection & playerSelection at function scope + pass to other functions as args

Declared the variables computerSelection and playerSelection locally at function scope. These were then passed to the respective functions which needed them as arguments.

Calerid committed:
Moved addWin functions below comparison functions

Moved the three add score functions down below the RPS comparison functions. This was done to make reading the program a more linear process. It does not currently serve any programming changes

Calerid committed:
Remove global playerSelection declaration

playerSelection is no longer globally scoped and is now declared and initialized in the set player functions. This was an uneeded left over from a previous refactor.

Calerid committed:
Remove 'else if' in playGame

Removed uneeded else if statement within playGame(). This will not occur within the normal flow of the game any longer and is therfore uneeded.

Calerid committed:
Remove redundancy in set functions

Removed setRock(), setPaper(), and setScissor() functions. Also removed computerPlay() function. This was done to reduce redundant code. Buttons now call playGame() with their respective values being called as arguments. computerPlay() was removed and the variable is now declared and assigned in playGame(). This reduces the need for a seperate function.
